js/utilities.js

```diff
@@ -96,6 +96,90 @@ var ERMrest = (function(module) {
         }
     };
 
+    /**
+     *
+     * @param http
+     * @param q
+     * @param method
+     * @param url
+     * @param data
+     * @returns {Promise}
+     * @private
+     */
+    module._makeRequest = function(http, q, method, url, data) {
```

Comment:
I think the API here should be a wrapper around the `http` service and not a new api. So you would implement these methods as wrappers: `http.get | .put | .post | .delete(...)`.

Also there's no need for the caller to pass in the `http` and `q` anyway because the module already has references to them. But with my suggested change to the api (as a wrapper on http) that goes away too.